innertube/src/main/java/com/zionhuang/innertube/models/response/PlayerResponse.kt

```diff
@@ -57,9 +60,25 @@ data class PlayerResponse(
             val audioChannels: Int?,
             val loudnessDb: Double?,
             val lastModified: Long?,
+            val signatureCipher: String?,
         ) {
             val isAudio: Boolean
                 get() = width == null
+
+            fun findUrl(): String? {
+                this.url?.let {
+                    return it
+                }
+                this.signatureCipher?.let { signatureCipher ->
+                    val params = parseQueryString(signatureCipher)
+                    val obfuscatedSignature = params["s"] ?: return null
+                    val signatureParam = params["sp"] ?: return null
+                    val url = params["url"]?.let { URLBuilder(it) } ?: return null
+                    url.parameters[signatureParam] = YoutubeJavaScriptPlayerManager.deobfuscateSignature("", obfuscatedSignature)
+                    return YoutubeJavaScriptPlayerManager.getUrlWithThrottlingParameterDeobfuscated("", url.toString())
```

Comment:
Same as before for the `videoId` being `""`. Here it's not the same since it's not passed in parameter somewhere near, so I guess it's good for now.